docs/remove-specified-tags-from-all-products/script.liquid

```diff
@@ -0,0 +1,84 @@
+{% assign tags_list = options.tags_to_remove__required_string | split: "," %}
+{% assign tagged_product_ids = array %}
+{% assign tags_query = "" %}
+{% for tag in tags_list %}
+  {% assign stripped_tag = tag | strip %}
+  {% assign tags_query = tags_query | append: "(tag:" | append: stripped_tag | append: ")" %}
```

Comment:
Sometimes tags have spaces in them - or parentheses. :) How would you like to handle this?

If it helps, there's a pattern in the task library that could be useful:

https://github.com/lightward/mechanic-tasks/blob/e7a623fb3b978ed36061f2c92353aff21766ed4e/docs/auto-tag-customers-based-on-discount-codes-used/script.liquid#L76-L80

Using the `json` filter wraps the value in double quotes, and escapes any necessary characters. Prepending `"tag:"` (which is what you'd do for your task, instead of `"discount_code:`) puts your search attribute where it belongs.

Reply:
Much nicer! Thanks @isaacbowen I've pushed that refactor.